deployability/modules/allocation/vagrant/utils.py

```diff
@@ -60,25 +66,40 @@ def remote_copy(cls, instance_dir: Path, host_instance_dir: Path, macos_host_par
                                         stderr=subprocess.PIPE)
             _, stderr = output.communicate()
             if stderr:
-                logger.error(f"Command failed: {stderr.decode('utf-8')}")
-                return None
+                raise ValueError(f"Command failed: {stderr.decode('utf-8')}")
             return output.stdout
         except subprocess.CalledProcessError as e:
-            logger.error(f"Command failed: {e.stderr.decode('utf-8')}")
-            return None
+            raise ValueError(f"Command failed: {e.stderr.decode('utf-8')}")
 
     @classmethod
-    def get_port(cls, macos_host_parameters: dict) -> int:
+    def get_port(cls, remote_host_parameters: dict, arch: str = None) -> int:
         """
         Returns the port for the remote host.
 
+        Args:
+            remote_host_parameters (dict): The parameters of the remote host.
+            arch (str): The architecture of the remote host.
+
         Returns:
             int: The port.
         """
 
-        for i in range(20, 40):
-            port = f"432{i}"
-            cmd = f"sudo lsof -i:{port}"
-            output = cls.remote_command(cmd, macos_host_parameters)
+        if arch == 'ppc64':
+            cmd = "sudo lsof -i:8080"
+            output = cls.remote_command(cmd, remote_host_parameters)
             if not output:
-                return port
+                return str(8080)
+            else:
+                cmd = "sudo lsof -i:2222"
+                output = cls.remote_command(cmd, remote_host_parameters)
+                if not output:
+                    return str(2222)
+                else:
+                    raise ValueError(f"ppc64 server has no available SSH ports.")
+        else:
+            for i in range(20, 40):
+                port = f"432{i}"
+                cmd = f"sudo lsof -i:{port}"
+                output = cls.remote_command(cmd, remote_host_parameters)
+                if not output:
+                    return port
```

Comment:
It could be "simplified" by removing some elses in the cases where the `if` does a return

```suggestion
                return str(8080)

            cmd = "sudo lsof -i:2222"
            output = cls.remote_command(cmd, remote_host_parameters)
            if not output:
                return str(2222)

            raise ValueError(f"ppc64 server has no available SSH ports.")
        else:
            for i in range(20, 40):
                port = f"432{i}"
                cmd = f"sudo lsof -i:{port}"
                output = cls.remote_command(cmd, remote_host_parameters)
                if not output:
                    return port
```

Reply:
Done: dc8184857